src/lock.test.js

```diff
@@ -98,3 +100,16 @@ test("sequentialized", async () => {
 
   expect(arr).toStrictEqual([100, 10]);
 });
+
+test("throttle", async () => {
+  let factor = 0;
+
+  const multiply = async (x) => {
+    factor = factor + 1;
+    await sleep(0.01);
+    return x * factor;
+  };
+
+  const result = await asyncMap(throttle(1, multiply))([1, 2, 3]);
```

Comment:
I think the test can be more direct and not rely on race conditions that in theory can work. what do you think about this:
```
let maxConcurrent = 0;
let insideNow = 0;

const in = () => {
  insideNow ++;
  maxConcurrent = maxConcurrent > insideNow ? maxConcurrent : insideNow;
}

const out = () => {
  inNow--;
}
```

then i'd `expect(maxConcurrent).toEqual(3)` without `throttle` and `expect(maxConcurrent).toEqual(1)` with.

Reply:
```
const multiply = async (x) => {
    in();
    await sleep(0.01);
    out();
    return x;
  };
```

Reply:
will work but I don't understand the difference ? you are relying on the same race conditions